contracts/TokenFactory.sol

```diff
@@ -27,15 +31,20 @@ contract TokenFactory is ReentrancyGuard, Ownable {
     mapping(uint => address) private tokensAddresses;
     uint totalTokensAddresses;
 
+    mapping(address => uint256) public creationDate;
```

Comment:
could be `address => competition id` instead, if we are to add a liquidity threshold for concluding each competition (thus each competition doesn't have to finish within `maxFundingRateInterval`)